Fatal Error during make build

Following the guide explicitly copy and pasting all commands.
Here is theterminal dump:

[ 44%] Building CXX object src/supernode/CMakeFiles/obj_supernode.dir/PosSaleObject.cpp.o
/home/tachy/GraftNetwork/src/supernode/PosSaleObject.cpp:34:23**: fatal error: uuid/uuid.h: No such file or directory**
compilation terminated.
src/supernode/CMakeFiles/obj_supernode.dir/build.make:374: recipe for target 'src/supernode/CMakeFiles/obj_supernode.dir/PosSaleObject.cpp.o' failed
make[3]: *** [src/supernode/CMakeFiles/obj_supernode.dir/PosSaleObject.cpp.o] Error 1
make[3]: Leaving directory '/home/tachy/GraftNetwork/build/release'
CMakeFiles/Makefile2:1413: recipe for target 'src/supernode/CMakeFiles/obj_supernode.dir/all' failed
make[2]: *** [src/supernode/CMakeFiles/obj_supernode.dir/all] Error 2
make[2]: Leaving directory '/home/tachy/GraftNetwork/build/release'
Makefile:138: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/tachy/GraftNetwork/build/release'
Makefile:62: recipe for target 'release-all' failed
make: *** [release-all] Error 2

--- a/README.md
+++ b/README.md
@@ -89,22 +89,23 @@ sources are also used for statically-linked builds because distribution
 packages often include only shared library binaries (`.so`) but not static
 library archives (`.a`).
 
-| Dep            | Min. Version  | Vendored | Debian/Ubuntu Pkg  | Arch Pkg       | Optional | Purpose        |
-| -------------- | ------------- | ---------| ------------------ | -------------- | -------- | -------------- |
-| GCC            | 4.7.3         | NO       | `build-essential`  | `base-devel`   | NO       |                |
-| CMake          | 3.0.0         | NO       | `cmake`            | `cmake`        | NO       |                |
-| pkg-config     | any           | NO       | `pkg-config`       | `base-devel`   | NO       |                |
-| Boost          | 1.58          | NO       | `libboost-all-dev` | `boost`        | NO       | C++ libraries  |
-| OpenSSL        | basically any | NO       | `libssl-dev`       | `openssl`      | NO       | sha256 sum     |
-| libunbound     | 1.4.16        | YES      | `libunbound-dev`   | `unbound`      | NO       | DNS resolver   |
-| libminiupnpc   | 2.0           | YES      | `libminiupnpc-dev` | `miniupnpc`    | YES      | NAT punching   |
-| libunwind      | any           | NO       | `libunwind8-dev`   | `libunwind`    | YES      | Stack traces   |
-| liblzma        | any           | NO       | `liblzma-dev`      | `xz`           | YES      | For libunwind  |
-| ldns           | 1.6.17        | NO       | `libldns-dev`      | `ldns`         | YES      | SSL toolkit    |
-| expat          | 1.1           | NO       | `libexpat1-dev`    | `expat`        | YES      | XML parsing    |
-| GTest          | 1.5           | YES      | `libgtest-dev`^    | `gtest`        | YES      | Test suite     |
-| Doxygen        | any           | NO       | `doxygen`          | `doxygen`      | YES      | Documentation  |
-| Graphviz       | any           | NO       | `graphviz`         | `graphviz`     | YES      | Documentation  |
+| Dep            | Min. Version  | Vendored | Debian/Ubuntu Pkg  | Arch Pkg       | Optional | Purpose                       |
+| -------------- | ------------- | ---------| ------------------ | -------------- | -------- | ----------------------------- |
+| GCC            | 4.7.3         | NO       | `build-essential`  | `base-devel`   | NO       |                               |
+| CMake          | 3.0.0         | NO       | `cmake`            | `cmake`        | NO       |                               |
+| pkg-config     | any           | NO       | `pkg-config`       | `base-devel`   | NO       |                               |
+| Boost          | 1.58          | NO       | `libboost-all-dev` | `boost`        | NO       | C++ libraries                 |
+| OpenSSL        | basically any | NO       | `libssl-dev`       | `openssl`      | NO       | sha256 sum                    |
+| libunbound     | 1.4.16        | YES      | `libunbound-dev`   | `unbound`      | NO       | DNS resolver                  |
+| uuid           | any           | NO       | `uuid-dev`         | `uuid`         | NO       | Universally Unique ID library |
+| libminiupnpc   | 2.0           | YES      | `libminiupnpc-dev` | `miniupnpc`    | YES      | NAT punching                  |
+| libunwind      | any           | NO       | `libunwind8-dev`   | `libunwind`    | YES      | Stack traces                  |
+| liblzma        | any           | NO       | `liblzma-dev`      | `xz`           | YES      | For libunwind                 |
+| ldns           | 1.6.17        | NO       | `libldns-dev`      | `ldns`         | YES      | SSL toolkit                   |
+| expat          | 1.1           | NO       | `libexpat1-dev`    | `expat`        | YES      | XML parsing                   |
+| GTest          | 1.5           | YES      | `libgtest-dev`^    | `gtest`        | YES      | Test suite                    |
+| Doxygen        | any           | NO       | `doxygen`          | `doxygen`      | YES      | Documentation                 |
+| Graphviz       | any           | NO       | `graphviz`         | `graphviz`     | YES      | Documentation                 |
 
 [^] On Debian/Ubuntu `libgtest-dev` only includes sources and headers. You must
 build the library binary manually. This can be done with the following command ```sudo apt-get install libgtest-dev && cd /usr/src/gtest && sudo cmake . && sudo make && sudo mv libg* /usr/lib/ ```

--- a/install_dependencies.sh
+++ b/install_dependencies.sh
@@ -7,6 +7,7 @@ sudo apt-get install -y \
     libboost-all-dev \
     libssl-dev \
     libunbound-dev \
+    uuid-dev \
     libminiupnpc-dev \
     libunwind8-dev \
     liblzma-dev \

